test/clientlib.js

```diff
@@ -102,6 +103,39 @@ contract('RelayClient', function (accounts) {
 
     });
 
+    it("should consider a transaction with a relay tx nonce higher than expected as invalid", async function () {
+        let encoded = sr.contract.methods.emitMessage("hello world").encodeABI()
+        let to = sr.address;
+        let options = {
+            from: gasLess,
+            to: to,
+            txfee: 12,
+            gas_limit: 1000000
+        }
+        let relay_client_config = {
+            relayUrl: localhostOne,
+            relayAddress: relayAddress,
+            allowed_relay_nonce_gap: -1
+        }
+        let tbk = new RelayClient(web3, relay_client_config);
+        let orig_send = tbk.httpSend.send
+        tbk.httpSend.send = function(url, jsonRequestData, callback){
+            if (url.includes("/relay")) {
+                // Otherwise, server will return an error if asked to sign with a low nonce.
+                jsonRequestData.RelayMaxNonce = 1000000
+            }
+            orig_send.bind(tbk.httpSend)(url, jsonRequestData, callback)
+        }
+        try {
+            await tbk.relayTransaction(encoded, options);
+            assert.fail()
+        }
+        catch(error) {
+            assert.equal(true, error.otherErrors[0].includes("Relay used a tx nonce higher than requested"))
```

Comment:
doesn't not handle assert.fail good.
probably the fix is 
{{if (error.instanceOf(AssertionError) throw e;}}